### PR TITLE
feat: azure provider

### PR DESCRIPTION
**Pull request by AlexcastroDev**
Only approve after merge after: https://github.com/ryo-ma/github-profile-trophy/pull/230

# Feature

This PR provides a private provider for the data with Redis caching

## Reason

Currently we decrease the request to Github API, but /tmp is deleted when machine ends working. ( Serveless reasons )

## What's new ?

The Azure API will retain Github Response for some hours, even /tmp folder delete file. 

## PS

it will not change after merge, only when we change the provider in vercel enviroments

## Testing

Link for testing: https://github-profile-trophy-git-feat-azure-alexcastrodevbr.vercel.app/?username=ryo-ma

### CODE REVIEW

**Comment by ryo-ma** [comment]
Please rebase

File: src/StaticRenderRegeneration/utils.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function readCache(cacheFilePath: string): Uint8Array | null {
 }
 
 // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
-export async function hashString(message: string): Promise<string> {
+export async function generateUUID(message: string): Promise<string> {
```
Comment (ryo-ma) [inline src/StaticRenderRegeneration/utils.ts:21]: Did you change the name back to UUID?

File: src/Services/GithubAzureService.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    ]);
+    const [activity, issue, pullRequest] = await promises;
+    const status = [
+      activity.status,
+      issue.status,
+      pullRequest.status,
+    ];
+
+    if (status.includes("rejected")) {
+      console.error(`Can not find a user with username:' ${username}'`);
```
Comment (ryo-ma) [inline src/Services/GithubAzureService.ts:82]: Don't you use Logger?

File: src/Services/GithubAzureService.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  async executeQuery<T = unknown>(
+    query: string,
+    variables: { [key: string]: string },
+    cache_ns?: string,
+  ) {
+    const retry = new Retry(
+      TOKENS.length,
+      CONSTANTS.DEFAULT_GITHUB_RETRY_DELAY,
+    );
+    console.log("HIT");
```
Comment (ryo-ma) [inline src/Services/GithubAzureService.ts:103]: Don't you use Logger?

File: src/Services/GithubAzureService.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+          },
+        });
+      }) as QueryDefaultResponse<{ data: { user: T } }>;
+
+      return response?.data?.data?.data?.user ??
+        new ServiceError("not found", EServiceKindError.NOT_FOUND);
+    } catch (error) {
+      // TODO: Move this to a logger instance later
+      if (error instanceof Error && error.cause) {
+        console.error(JSON.stringify(error.cause, null, 2));
```
Comment (ryo-ma) [inline src/Services/GithubAzureService.ts:120]: Don't you use Logger?

File: src/Services/GithubAzureService.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      } else {
+        console.error(error);
```
Comment (ryo-ma) [inline src/Services/GithubAzureService.ts:122]: Same

File: src/Services/GithubAzureService.ts (hunk 3 above)
Comment (AlexcastroDev) [reply src/Services/GithubAzureService.ts]: removed

File: src/Services/GithubAzureService.ts (hunk 2 above)
Comment (AlexcastroDev) [reply src/Services/GithubAzureService.ts]: fixed

File: src/StaticRenderRegeneration/utils.ts (hunk 1 above)
Comment (AlexcastroDev) [reply src/StaticRenderRegeneration/utils.ts]: fixed

File: src/Services/GithubAzureService.ts (hunk 4 above)
Comment (AlexcastroDev) [reply src/Services/GithubAzureService.ts]: fixed

File: src/Services/GithubAzureService.ts (hunk 5 above)
Comment (AlexcastroDev) [reply src/Services/GithubAzureService.ts]: fixed

**Comment by AlexcastroDev** [comment]
Testing link: https://github-profile-trophy-git-feat-azure-ryo-ma-s-team.vercel.app/?username=ryo-ma

**Pull request closed without merging**